public/app/features/alerting/unified/components/silences/SilencesEditor.tsx

```diff
@@ -24,22 +33,27 @@ const getDefaultFormValues = (silence?: Silence): SilenceFormFields => {
   if (silence) {
     return {
       id: silence.id,
-      startsAt: new Date().toISOString(),
-      endsAt: new Date(Date.now() + 2 * 60 * 60 * 1000).toISOString(), // Default time period is now + 2h
+      startsAt: silence.startsAt,
+      endsAt: silence.endsAt,
       comment: silence.comment,
       createdBy: silence.createdBy,
-      duration: `2h`,
+      duration: intervalToAbbreviatedDurationString({
+        start: new Date(silence.startsAt),
```

Comment:
even when recreating the silence, doesn't make sense to use it's period - it's in the past. So in this case as well let's use 2h duration